src/ast/semantic_analyser.cpp

```diff
@@ -559,6 +559,17 @@ void SemanticAnalyser::visit(Call &call)
   else if (call.func == "ustack") {
     check_stack_call(call, Type::ustack);
   }
+  else if (call.func == "strncmp") {
+    check_nargs(call, 3);
+    check_arg(call, Type::string, 0);
+    check_arg(call, Type::string, 1);
```

Comment:
Can you add a few semantic analyser tests for this? Especially the "wrong" cases are good to test, e.g.:

```
TEST(semantic_analyser, strncmp)
{
  test("i:s:1 { strncmp(1) }", 1);
  test("i:s:1 { strncmp(1,1,1) }", 1);
  test("i:s:1 { strncmp("a",1,1) }", 1);
  test("i:s:1 { strncmp("a","a",-1) }", 1);
}
```